src/scoped.rs

```diff
@@ -264,6 +352,21 @@ impl<'a> Scope<'a> {
         });
     }
 
+    /// Schedule code to be executed when exiting the scope.
```

Comment:
Is this description correct? It seems this method is specialized for deferring thread join actions rather than generic functions.

Reply:
`DtorChain` is used not only for joins but also arbitrary dtors: https://github.com/jeehoonkang/crossbeam-utils/blob/scope-join-returns-result/src/scoped.rs#L153-L154

Reply:
Yes, but the specific method `defer_join` is only used for deferring functions that return `Option<thread::Result<()>>`, so that seems like it's meant for threads.